neurom/features/morphology.py

```diff
@@ -55,11 +55,60 @@
 from neurom.core.types import NeuriteType
 from neurom.features import feature, NameSpace, neurite as nf
 from neurom.utils import str_to_plane
-
+from neurom.core.morphology import Neurite
+from collections import OrderedDict
 
 feature = partial(feature, namespace=NameSpace.NEURON)
 
 
+def _homogeneous_subtrees(neurite):
+    """Returns a dictionary the keys of which are section types and the values are the
+    sub-neurites. A sub-neurite can be either the entire tree or a homogeneous downstream
+    sub-tree.
+
+    Note: Only two different mixed types are allowed
+    """
+    homogeneous_neurites = OrderedDict([(neurite.root_node.type, neurite)])
+    for section in neurite.root_node.ipreorder():
+        if section.type not in homogeneous_neurites:
+            homogeneous_neurites[section.type] = Neurite(section)
+    if len(homogeneous_neurites) != 2:
+        raise TypeError(
+            f"Subtree types must be exactly two. Found {len(homogeneous_neurites)} instead.")
```

Comment:
is this an invariant we need? could it be more?

I've been asking myself if we should just handle the AoD case, or make a more general solution; I have no idea what biologically makes sense, though.